File: frontend/src/types/icp.ts
export interface ICPDemographics {
  age_range: string;
  gender: string;
  income_level: string;
  education: string;
  location: string;
  occupation: string;
}

export interface ICPPsychographics {
  values: string[];
  interests: string[];
  lifestyle: string;
  media_consumption: string[];
  personality_traits: string[];
}

export interface ICPProfile {
  demographics: ICPDemographics;
  psychographics: ICPPsychographics;
  pain_points: string[];
  desires: string[];
  objections: string[];
  language_style: string;
}

export interface ICPProfileResponse {
  id: string;
  project_id: string;
  demographics: ICPDemographics;
  psychographics: ICPPsychographics;
  pain_points: string[];
  desires: string[];
  objections: string[];
  language_style: string;
  source: string;
  approved: boolean;
  created_at: string;
  updated_at: string;
}

export interface ICPUpdateRequest {
  demographics?: ICPDemographics | null;
  psychographics?: ICPPsychographics | null;
  pain_points?: string[] | null;
  desires?: string[] | null;
  objections?: string[] | null;
  language_style?: string | null;
  approved?: boolean | null;
}

File: backend/app/schemas/icp.py
from pydantic import BaseModel, Field


class ICPDemographics(BaseModel):
    age_range: str = Field(default="", description="Target age range")
    gender: str = Field(default="", description="Target gender")
    income_level: str = Field(default="", description="Target income level")
    education: str = Field(default="", description="Target education level")
    location: str = Field(default="", description="Target location")
    occupation: str = Field(default="", description="Target occupation")


class ICPPsychographics(BaseModel):
    values: list[str] = Field(default_factory=list, description="Core values")
    interests: list[str] = Field(default_factory=list, description="Key interests")
    lifestyle: str = Field(default="", description="Lifestyle description")
    media_consumption: list[str] = Field(default_factory=list, description="Media consumption habits")
    personality_traits: list[str] = Field(default_factory=list, description="Personality traits")


class ICPProfile(BaseModel):
    demographics: ICPDemographics = Field(default_factory=ICPDemographics)
    psychographics: ICPPsychographics = Field(default_factory=ICPPsychographics)
    pain_points: list[str] = Field(default_factory=list)
    desires: list[str] = Field(default_factory=list)
    objections: list[str] = Field(default_factory=list)
    language_style: str = Field(default="professional")


class ICPAgentInput(BaseModel):
    raw_notes: str = ""
    topic: str | None = None
    target_format: str | None = None
    content_goal: str | None = None
    piragi_context: str | None = None


class ICPAgentOutput(BaseModel):
    icp: ICPProfile
    confidence: float = Field(ge=0, le=1)


class ICPProfileResponse(BaseModel):
    id: str
    project_id: str
    demographics: ICPDemographics
    psychographics: ICPPsychographics
    pain_points: list[str]
    desires: list[str]
    objections: list[str]
    language_style: str
    source: str
    approved: bool
    created_at: str
    updated_at: str

    model_config = {"from_attributes": True}


class ICPUpdateRequest(BaseModel):
    demographics: ICPDemographics | None = None
    psychographics: ICPPsychographics | None = None
    pain_points: list[str] | None = None
    desires: list[str] | None = None
    objections: list[str] | None = None
    language_style: str | None = None
    approved: bool | None = None
